### PR TITLE
WIP Initial ovsdb-etcd ovnkube-master yaml

### PR DESCRIPTION
**Pull request by msherif1234**
Signed-off-by: Mohamed Mahmoud <mmahmoud@redhat.com>

### CODE REVIEW

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 1]
```diff
@@ -103,331 +97,68 @@ spec:
...
-          # Checks if cluster has already been initialized.
-          # If not it returns false and sets init_ip to CLUSTER_INITIATOR_IP
+          set -x
           cluster_exists() {
-            local db=${1}
-            local port=${2}
-            # TODO: change to use '--request-timeout=5s', if https://github.com/kubernetes/kubernetes/issues/49343 is fixed. 
-            db_pods=$(timeout 5 kubectl get pod -n ${ovn_kubernetes_namespace} -o=jsonpath='{.items[*].metadata.name}' | egrep -o 'ovnkube-master-\w+' | grep -v "metrics")
+           ovsdb_etcd_members=${OVSDB_ETCD_MEMBERS:-"localhost:2479"}
+           ovn_sb_port=${OVN_NB_PORT:-664}
```
Comment (roytman) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml:109]: OVN_NB_PORT - > OVN_SB_PORT
the default should be 6642, and probably we should set it to `{{.OVN_SB_PORT}}`

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 2]
```diff
@@ -103,331 +97,68 @@ spec:
             memory: 300Mi
         terminationMessagePolicy: FallbackToLogsOnError
 
-      # nbdb: the northbound, or logical network object DB. In raft mode 
-      - name: nbdb
-        image: "{{.OvnImage}}"
+      - name: sb-ovsdb
```
Comment (roytman) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml:100]: looks like a mismatch between original NB DB and SB ovsdb-etcd

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 3]
```diff
@@ -103,331 +97,68 @@ spec:
...
-            set -o allexport
-            source /env/_master
-            set +o allexport
-          fi
-
-          bracketify() { case "$1" in *:*) echo "[$1]" ;; *) echo "$1" ;; esac }
-          # initialize variables
-          ovn_kubernetes_namespace=openshift-ovn-kubernetes
-          ovndb_ctl_ssl_opts="-p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt"
-          transport="ssl"
```
Comment (roytman) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml:124]: we have not added SSL support to ovsdb-etcd.
Therefore we have 2 options:
add SSL support to ovsdb-etcd or to configure all ovsdb clients to work over plain TCP protocol.

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 4]
```diff
@@ -480,291 +211,135 @@ spec:
           mountPath: /etc/pki/tls/metrics-cert
           readOnly: True
 
-      # sbdb: The southbound, or flow DB. In raft mode 
-      - name: sbdb
-        image: "{{.OvnImage}}"
+      - name: nb-ovsdb
```
Comment (roytman) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml:214]: mismatch of SB and NB again

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 5]
```diff
@@ -480,291 +211,135 @@ spec:
...
+      - name: etcd
+        image: "quay.io/mmahmoud/etcd:latest"
+        imagePullPolicy: "{{ ovn_image_pull_policy | default('IfNotPresent') }}"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -x
+          # OVSDB-etcd variables
+          ovsdb_etcd_max_txn_ops=${OVSDB_ETCD_MAX_TXN_OPS:-"157286400"}                   # etcd default is 128
```
Comment (roytman) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml:287]: 150M is too much, let's reduce it to 150K

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 6]
```diff
@@ -103,331 +97,68 @@ spec:
...
-          
-          # Checks if cluster has already been initialized.
-          # If not it returns false and sets init_ip to CLUSTER_INITIATOR_IP
+          set -x
           cluster_exists() {
-            local db=${1}
-            local port=${2}
-            # TODO: change to use '--request-timeout=5s', if https://github.com/kubernetes/kubernetes/issues/49343 is fixed. 
-            db_pods=$(timeout 5 kubectl get pod -n ${ovn_kubernetes_namespace} -o=jsonpath='{.items[*].metadata.name}' | egrep -o 'ovnkube-master-\w+' | grep -v "metrics")
+           ovsdb_etcd_members=${OVSDB_ETCD_MEMBERS:-"localhost:2479"}
```
Comment (roytman) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml:108]: This configuration and the similar one for NB as well etcd server start arguments assume that there is a single etcd server and SB and NB are colocated with it in the same Pod. 

Will OCP run 3 master nodes with similar config on each of them ?
Should we start from a single master and after that change the configuration to support etcd cluster and several or dedicated ovsdb-etcd servers.

**Review by roytman — COMMENTED** [review]
Hi @msherif1234 , I added several small comments. 
The main challenges are: SSL and cluster support (slightly different configuration). 

We can validate deployment with a single none-SSL server and add the missing functionality after that.

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml (hunk 1 above)
Comment (msherif1234) [reply bindata/network/ovn-kubernetes/ovnkube-master.yaml:109]: opps typo I dropped 2 at the end

**Pull request closed without merging**